protocol/superchainERC20/convert.md

```diff
@@ -0,0 +1,176 @@
+
+However, a manual migration could take months, and a forced migration would introduce significant liability. For these reasons, we advise against this path.
+
+**L2 migration**
+
+Another approach is to make changes only on the L2 side to make the current `OptimismMintableERC20` tokens interoperable without many protocol-level changes. This approach is simpler and safer than L1 migration.
+
+There are several ways to achieve this result:
+
+- **Wrap:** Introduce a new contract that wraps one or multiple legacy tokens into a `SuperchainERC20`.
+- **Convert:** Use the burn/mint rights of the `L2StandardBridge` and allow users to convert back and forth between the `OptimismMintableERC20` and the corresponding `SuperchainERC20`.
+- **Mirror:** Create a `SuperchainERC20` representation that behaves (with some exceptions) like the legacy "mirrored" token. They will act as secondary entry points to each other.
+
+**Tradeoffs**
+
+- The Wrap approach depends on locked liquidity. This can generate some friction for withdrawals, as `SuperchainERC20`s will move across chains while liquidity remains locked.
+  In contrast, the Convert approach uses mint and burn rights for flexible liquidity exchange between standards.
+- Convert and Wrap require user migration to be interop compatible. On the other hand, Mirror needs no action and will work from moment zero.
+- The Mirror approach might bring security implications at the application layer, as already happened with double entry point designs. Wrap and Convert are very well-known designs that should not alter the existing applications.
+
+> ✅ Of all three methods, the Convert solution represents the cleanest solution to transform existing liquidity into `SuperchainERC20`. This is because of the balance between UX and safety.
+
+### Isolated vs Unified
+
+The Convert method will accept `OptimismMintableERC20`s and convert them into `SuperchainERC20`s. The most significant decision is whether to have a single `SuperchainERC20` that accepts multiple `OptimismMintableERC20`s of the same L1 token (Unified) or to maintain one `SuperchainERC20` per `OptimismMintableERC20` (Isolated).
+
+We prefer simplicity and will therefore focus on the Isolated approach, i.e., a single `SuperchainERC20` per legacy token. For a more detailed discussion, please refer to the Appendix.
+
+# Proposed Solution: Isolated Convert
+
+## Implementation
+
+The Convert Method will require modifying the `L2StandardBridge` to include the following:
+
+- `convertToSuper(address _*legacyAddr, address _superAddr, uint256 _amount)`: converts* `_amount` *legacy tokens with address `_*legacyAddr` to the same `_amount` of `SuperchainERC20` tokens in address `_superAddr`.
```

Comment:
Sanity check: is the `_legacyAddr` the same across all L2 chains, for the same L1 token?